src/main/java/core/basesyntax/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    private static final String INPUT_DATA = "fruit_shop_transactions.csv";
+
+    @Override
+    public List<FruitTransaction> readFile(String fileName) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new java.io.FileReader(fileName))) {
+            String line;
+            br.readLine();
+            while ((line = br.readLine()) != null) {
+                String[] data = line.split(",");
+                OperationType operation = OperationType.fromCode(data[0]);
+                String fruit = data[1];
+                int quantity = Integer.parseInt(data[2]);
```

Comment:
Consider adding validation for the parsed data. For instance, check if the operation code is valid, if the fruit name is not empty, and if the quantity is non-negative. This will make your application more robust and user-friendly.